tests/mq/plugins/test_broFixup.py

```diff
@@ -0,0 +1,1697 @@
+            'index': 'events'
+        }
+        event = {
+           'category': 'bro'
+        }
+       
+        result, metadata = self.plugin.onMessage(event, metadata)
+        assert result
+        assert metadata['doc_type'] is not 'nsm'
+    
+    def test_bro_wrongtype_log(self):
+        event = {
+           'category': 'bro',
+           'type': 'nosuchtype',
+           'ts': 1505701210.163043
+        }
+       
+        result, metadata = self.plugin.onMessage(event, self.metadata)
+        self.verify_defaults(result)
+        self.verify_metadata(metadata)
+        assert toUTC(event['ts']).isoformat() == result['utctimestamp']
+        assert toUTC(event['ts']).isoformat() == result['timestamp']
+        assert sorted(result['details'].keys()) == sorted(event.keys())
+        assert metadata['doc_type'] is 'nsm'
+
+    @mock.patch('broFixup.node')
+    def test_mozdefhostname_mock_string(self, mock_path):
+        mock_path.return_value = 'samplehostname'
+        event = {
+            'category': 'bro',
+            'type': 'something'
+        }
+        plugin = message()
+        result, metadata = plugin.onMessage(event, self.metadata)
+        assert result['mozdefhostname'] == 'samplehostname'
+
+    @mock.patch('broFixup.node')
+    def test_mozdefhostname_mock_exception(self, mock_path):
+        mock_path.side_effect = ValueError
+        event = {
+            'category': 'bro',
+            'type': 'something'
+        }
+        plugin = message()
+        result, metadata = plugin.onMessage(event, self.metadata)
+        assert result['mozdefhostname'] == 'failed to fetch mozdefhostname'
+
+    def verify_metadata(self, metadata):
+        assert metadata['doc_type'] == 'nsm'
+
+    def test_defaults(self):
+        event = {
+            'category': 'bro',
+            'type': 'something'
+        }
+        result, metadata = self.plugin.onMessage(event, self.metadata)
+        self.verify_defaults(result)
+        self.verify_metadata(metadata)
+        assert result['category'] == 'bro'
+        assert result['type'] == 'something'
+    
+    def test_nomatch_syslog(self):
+        event = {
+                "category": "syslog",
+                "processid": "0",
+                "receivedtimestamp": "2017-09-26T00:22:24.210945+00:00",
+                "severity": "7",
+                "utctimestamp": "2017-09-26T00:22:23+00:00",
+                "timestamp": "2017-09-26T00:22:23+00:00",
+                "hostname": "syslog1.private.scl3.mozilla.com",
+                "mozdefhostname": "mozdef1.private.scl3.mozilla.com",
+                "summary": "Connection from 10.22.74.208 port 9071 on 10.22.74.45 nsm bro port 22\n",
+                "eventsource": "systemslogs",
+                "details": {
+                    "processid": "21233",
+                    "Random": 2,
+                    "sourceipv4address": "10.22.74.208",
+                    "hostname": "hgssh4.dmz.scl3.mozilla.com",
```

Comment:
Do we want to use actual hostnames here, or should we replace mozilla specific values with generic ones?

Reply:
Changed.